homeassistant/components/unifi/__init__.py

```diff
@@ -0,0 +1,158 @@
+"""
+Support for devices connected to UniFi POE.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/unifi/
+"""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME, CONF_VERIFY_SSL)
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+
+from .const import (CONF_CONTROLLER, CONF_POE_CONTROL, CONF_SITE_ID,
+                    CONTROLLER_ID, DOMAIN, LOGGER)
+from .controller import UniFiController, get_controller
+from .errors import (
+    AlreadyConfigured, AuthenticationRequired, CannotConnect, UserLevel)
+
+DEFAULT_PORT = 8443
+DEFAULT_SITE_ID = 'default'
+DEFAULT_VERIFY_SSL = False
+
+REQUIREMENTS = ['aiounifi==3']
+
+
+async def async_setup(hass, config):
+    """Component doesn't support configuration through configuration.yaml."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the UniFi component."""
+    controller = UniFiController(hass, config_entry)
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+    controller_id = CONTROLLER_ID.format(
+        host=config_entry.data[CONF_CONTROLLER][CONF_HOST],
+        site=config_entry.data[CONF_CONTROLLER][CONF_SITE_ID]
+    )
+
+    if not await controller.async_setup():
+        return False
+
+    hass.data[DOMAIN][controller_id] = controller
+
+    if controller.mac is None:
+        return True
+
+    device_registry = await \
+        hass.helpers.device_registry.async_get_registry()
+    device_registry.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        connections={(CONNECTION_NETWORK_MAC, controller.mac)},
+        manufacturer='Ubiquiti',
```

Comment:
We should pass at least a model too. Preferably a name too.

Reply:
The name would be site id in that case, the model have to be defined manually, would be something like UniFi Controller

Reply:
UniFi Controller for both model and name = 👍